src/TimePicker.elm

```diff
@@ -243,6 +262,40 @@ update settings msg (TimePicker ({ value } as model)) =
             ( TimePicker { model | inputText = Nothing, value = updatedValue }, timeEvent )
 
 
+defaultHour : Settings -> Int
+defaultHour settings =
+    Maybe.withDefault 0 (find (settings.isHourDisabled >> not) allHours)
```

Comment:
I noticed this doesn't also look at the step applied. Doesn't it seem like it should follow the step? So if the minute step is 15, then it shouldn't default to `1` just because it wasn't disabled.

Reply:
Well the default is `0`. From my understanding if you have a 'step' setting it'll start from 0 and therefore `0` would be the first one enabled and selected.

Reply:
Sorry, I mean with the `find`! The `find` will be checking all hours, not the stepped hours. So if you set a step of 15 minutes then you shouldn't get a value that isn't on that step (unless there aren't any valid ones). Does that make sense?

Reply:
I see. See reworked implementation. Providing that function a list of `steppedTimeUnits`